ticket_upgrade_prediction/hyperparam_pipeline.py

```diff
@@ -0,0 +1,226 @@
+            self.optimize_hypers_using_grid_search(n_splits)
+        else:
+            raise ValueError(
+                "Unrecognizable searching_algo param. Currently available are: random, grid."
+            )
+
+    @staticmethod
+    def get_random_params(param_space: dict) -> dict:
+        return {
+            k: (random.choice(v) if type(v) == list else v)
+            for k, v in param_space.items()
+        }
+
+    def optimize_hypers_using_grid_search(self, n_splits: int) -> None:
+        for iteration_params in ParameterGrid(self.param_space):
+            self.append_params_and_calculate_scores(iteration_params, n_splits)
+
+    def append_params_and_calculate_scores(
+        self, iteration_params: dict, n_splits: int
+    ) -> None:
+        self.params.append(iteration_params)
+        score = self.create_splits_and_calc_scores(n_splits, iteration_params)
+        logger.info(f"score for params {iteration_params} -> {score}")
+        self.scores.append(score)
+
+    def optimize_hypers_using_random_search(
+        self, n_splits: int, n_iters: int
+    ) -> None:
+        for i in range(n_iters):
+            iteration_params = self.get_random_params(self.param_space)
+            self.append_params_and_calculate_scores(iteration_params, n_splits)
+
+    def create_splits_and_calc_scores(
+        self, n_splits: int, iteration_params: dict
+    ) -> np.ndarray:
+        kf = StratifiedKFold(n_splits) if self.stratify else KFold(n_splits)
+        return np.mean(
+            [
+                self.create_preds_for_hypers(
+                    train_index, test_index, iteration_params
+                )
+                for train_index, test_index in kf.split(self.X, self.y)
+            ]
+        )
+
+    def get_scaled_train_and_test_sets(
+        self, train_index: np.ndarray, test_index: np.ndarray
+    ) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+        scaler = StandardScaler()
+        X_train, X_test, y_train, y_test = (
+            self.X.iloc[train_index],
+            self.X.iloc[test_index],
+            self.y.iloc[train_index],
+            self.y.iloc[test_index],
+        )
+        X_train[self.cols_to_scale] = scaler.fit_transform(
+            X_train[self.cols_to_scale]
+        )
+        X_test[self.cols_to_scale] = scaler.transform(
+            X_test[self.cols_to_scale]
+        )
+        return X_train, X_test, y_train, y_test
+
+    def create_preds_for_hypers(
+        self,
+        train_index: np.ndarray,
+        test_index: np.ndarray,
+        iteration_params: dict,
+    ) -> float:
+        model = self.determine_model(iteration_params)
+        X_train, X_test, y_train, y_test = self.get_scaled_train_and_test_sets(
+            train_index, test_index
+        )
+        model.fit(X_train, y_train)
+        # predictions = model.predict_proba(X_test)[:, 1] if self.classification else model.predict(X_test)
+        predictions = model.predict(X_test)
```

Comment:
Why dont You use `Evaluator` class here?
I have updated `Metrics` dataclass here https://github.com/kuba1302/ticket-upgrade-prediction/pull/15, 
so it will be easier to choose metric from string.
You will be able to do something like this: 

```
evaluator = Evaluator(model=model, X=X_test, y=y_test)
metric = evaluator.get_all_metrics()
return metric.get_metric_from_string("roc_auc")
```

Reply:
reworked script to use Evaluator class